Validate mutually exclusive

akinsey opened the pull request.

akinsey committed:
feat(errors/custom_errors): allow custom message for invalid payload error

akinsey committed:
feat(helpers/validate): add function to check mutual exclusivity of body parameters

akinsey committed:
refactor(controller/post): update comment todos

akinsey committed:
test(helper/validate): add doctest for mutually_exclusive! function

akinsey committed:
docs(helpers/validate): fix spelling errors in docs

unenglishable reviewed (changes requested):
i take it back!

contains_all_keys is not a valid way to check that the keys are mutually exclusive

unenglishable commented:
taking over responsibility for correcting algo 👍

unenglishable committed:
test(helpers/validate): move docs to test (for testing later)

unenglishable committed:
test(helpers/validate): test mutually exclusive with several cases

unenglishable committed:
fix(helpers/validate): correctly implement mutually_exclusive

needs to check if attrs contains any two keys, not all keys

implement map_contains_any_two_keys_in_list

  uses guard clause for checking empty map
  (a match on %{} matches any map)

use map_contains_any_two_keys_in_list

unenglishable committed:
style(helpers/validate): mix format

unenglishable committed:
test(helpers/validate): mix format

akinsey committed:
Merge pull request #80 from epochtalk/validate-mutually-exclusive-fixes

Validate mutually exclusive fixes

unenglishable reviewed (approved):
👍 lets get that money
